### PR TITLE
Added Eq instance for GenericSignature

### PR DESCRIPTION
**Pull request by eskimor**
It compiles for me and the very basic tests pass.

### CODE REVIEW

File: test/Main.purs [hunk 1]
```diff
@@ -63,3 +66,8 @@ main = do
   print (toSignature (Proxy :: Proxy Foo))
   print (toSignature (Proxy :: Proxy (Array Foo)))
   print (toSignature (Proxy :: Proxy MyNewString))
+  log "Basic test of Eq GenericSignature instance:"
+  log $ "Foo == Foo:" <> show (toSignature (Proxy :: Proxy Foo) == toSignature (Proxy :: Proxy Foo))
```
Comment (paf31) [inline test/Main.purs]: I think `purescript-assert` would be good here, and we can add it as a dev dependency in `bower.json`.

File: src/Data/Generic.purs [hunk 2]
```diff
@@ -268,6 +268,24 @@ instance eqGeneric :: Eq GenericSpine where
...
 
+eqDataConstructor :: DataConstructor -> DataConstructor -> Boolean
+eqDataConstructor p1 p2 = p1.sigConstructor == p2.sigConstructor
+                       && zipAll (\x y -> x unit == y unit) p1.sigValues p2.sigValues
+
+instance eqGenericSignature :: Eq GenericSignature where
+  eq (SigProd s1 arr1) (SigProd s2 arr2) = s1 == s2
+                                        && length arr1 == length arr2
+                                        && zipAll eqDataConstructor arr1 arr2
+  eq (SigRecord arr1) (SigRecord arr2)   = zipAll (\x y -> x.recLabel == y.recLabel) arr1 arr2
```
Comment (paf31) [inline src/Data/Generic.purs]: I think we need to sort by `recLabel` here? The same goes for `Eq GenericSpine` actually, I suppose.

Comment (paf31) [inline src/Data/Generic.purs]: Also, this needs a length check.

Comment (eskimor) [inline src/Data/Generic.purs]: Ha- good catch, thank you!

Comment (eskimor) [inline src/Data/Generic.purs]: I sorted them according to their definition - no?

Comment (paf31) [inline src/Data/Generic.purs]: Well there's no guarantee they'll be in definition order in general. They might even come from types with the same labels and types but different orders. Or unification might reorder them.

Comment (eskimor) [inline src/Data/Generic.purs]: Ah - I misunderstood what you meant. Of course, this check is bogus - if the fields are in a different order! Fix is coming..

**Comment by garyb** [comment]
@paf31 how's this one now? I'll incorporate it in the RC if you're happy with it now.

**Comment by paf31** [comment]
:+1: Looks good to me, thanks!

**Pull request merged**